### ISSUE TITLE
Random maze generator

### ISSUE THREAD

**Issue by jensj12** [jensj12/MeesGame#30]
Take a look at maze generation and hopefully implement it.

- [ ] Only place one door of each color and put stuff intelligently behind it
- [ ] Generate some open fields

**Comment by jorisBarkema** [comment]
Some maze generating methods: http://stackoverflow.com/questions/38502/whats-a-good-algorithm-to-generate-a-maze

**Comment by jensj12** [comment]
We may need to agree on what kind of maze we want to generate first. Different generators generate mazes with different characteristics, so picking the correct algorithm is only our second step. My vote goes to:
 - Loops are allowed, but not many
 - The length of the path should be twice the largest dimension of the maze
 - Starting position should be uniform random
 - Parameters specified by the player are good to have

**Comment by jorisBarkema** [comment]
I agree parameters would be good, for example an option to choose if the starting position is random or not, for example always start at the bottom and go to the top. Not exactly sure what you mean with 'length of the path', is that the length of the shortest route from start to finish? 

Also, some issues, like whether loops are allowed or not, could be solved by making that a parameter as well, and choosing a different algorithm depending on what was chosen. 

Downside is that that would mean having to create several maze generating algorithms, and it might cluster the window, having too many options to choose from is confusing Upside is more variety in the mazes.
